stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -27,6 +31,25 @@ template <typename ReduceFunction, typename ReturnType, typename Vec,
           typename... Args>
 struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
                        Vec, Args...> {
+  struct scoped_args_tuple {
+    ScopedChainableStack stack_;
+    using args_tuple_t
+        = std::tuple<decltype(deep_copy_vars(std::declval<Args>()))...>;
+    std::unique_ptr<args_tuple_t> args_tuple_holder_;
+
+    template <typename... ArgsT>
+    explicit scoped_args_tuple(ArgsT&&... args_tuple)
+        : stack_(), args_tuple_holder_(nullptr) {
+      stack_.execute([&] {
+        args_tuple_holder_ = std::unique_ptr<args_tuple_t>(
+            new std::tuple<decltype(deep_copy_vars(args_tuple))...>(
+                deep_copy_vars(args_tuple)...));
+      });
+    }
+  };
+
+  using local_args_tuple_t = tbb::enumerable_thread_specific<scoped_args_tuple>;
```

Comment:
[optional]

Looking at the docs for this should this use the `arena_allocator` in core?

https://spec.oneapi.com/versions/latest/elements/oneTBB/source/thread_local_storage/enumerable_thread_specific_cls.html

Reply:
I don't think that is needed. The AD tape itself will still use the stan-math arena... but the small struct holding stack and tuple together doesn't really need the arena (maybe there is a benefit I am not aware of).